### PR TITLE
[4.1] AdminController - Add context if any, to redirect URL

### PR DESCRIPTION
**Pull request by pe7er**
Pull Request to **fix missing context** for **com_fields** when doing actions like publishing or unpublishing Field or Field Group items.

### Summary of Changes

In the **Custom Field Groups** and **Custom Fields** of Content & Users, the left menu closed after doing an action (like publish or unbublish) on a Custom Field Group Item or Custom Field Item.

#### **Field Groups** menu closes when doing actions
1. In back-end go to: Content > Field Groups
2. Create a Field group and Save & Close
field-groups-1
3. Notice the URL:
/administrator/index.php?option=com_fields&view=groups&context=com_content.article
4. Unpublish the Group
field-groups-2
5. Notice the new URL and the closed Menu:
/administrator/index.php?option=com_fields&view=groups
**The Context is not passed through, and the menu has been closed.** 
The same issue happens in other places with other Custom Fields too.

### Testing Instructions

Go to each of the following menu items and publish or unpublish an item:
1. Content > Field Groups /administrator/index.php?option=com_fields&view=groups&context=com_content.article
2. Content > Fields /administrator/index.php?option=com_fields&view=fields&context=com_content.article
3. Users > Field Groups /administrator/index.php?option=com_fields&view=groups&context=com_users.user
4. Users > Fields /administrator/index.php?option=com_fields&view=fields&context=com_users.user
/administrator/index.php?option=com_fields&view=fields&context=com_content.article

### Actual result BEFORE applying this Pull Request

After doing a publish or unpublish action in com_fields, the menu on the left closes because the URL is missing the ``&context=[component][view]``

### Expected result AFTER applying this Pull Request

**After doing a publish or unpublish action** in com_fields, the **menu** on **the left remains open** because the right context ``&context=[component][view]`` is passed through via the URL.

### CODE REVIEW

**Comment by brianteeman** [comment]
also see https://github.com/joomla/joomla-cms/issues/34067

File: libraries/src/MVC/Controller/AdminController.php [hunk 1]
```diff
@@ -115,6 +115,17 @@ public function __construct($config = array(), MVCFactoryInterface $factory = nu
 			}
 
 			$this->view_list = strtolower($r[2]);
+
+			// Add the context to $this->view_list to use in the redirect URL
+			if (isset($app->input))
```
Comment (laoneo) [inline libraries/src/MVC/Controller/AdminController.php]: App should always have an input

**Comment by joomdonation** [comment]
@pe7er The propose change does not look right to me. For this, you should modify code in [Fields](https://github.com/joomla/joomla-cms/blob/4.2-dev/administrator/components/com_fields/src/Controller/FieldsController.php) and [Groups](https://github.com/joomla/joomla-cms/blob/4.2-dev/administrator/components/com_fields/src/Controller/GroupsController.php) controllers of com_fields instead. Just add `getRedirectToListAppend` method to the two controllers with the below code should solve the issue:

```php
protected function getRedirectToListAppend()
{
	$context = $this->input->getString('context');

	if ($context)
	{
		return '&context=' . $context;
	}
}
```

**Comment by HLeithner** [comment]
This pull requests has automatically rebased to 4.2-dev.

**Comment by joomdonation** [comment]
Hi @pe7er 

Look like you are too busy and do not have time to make the suggested change as I commented https://github.com/joomla/joomla-cms/pull/37818#issuecomment-1166258787 . So I made the PR https://github.com/joomla/joomla-cms/pull/39926 to replace this one to have this issue fixed and close this PR. Hope that's fine for you. If you still want to finish this PR, please let me know. I will close mine and re-open yours. Thanks !

**Pull request closed without merging**